### PR TITLE
Test globus login

### LINKED ISSUE #98: Test login in Cypress with stub
We want to test that webapp behavior after login is correct. I think we want to use some combination of Cypress [stubs](https://docs.cypress.io/api/commands/stub.html) and mocked API responses. (Another approach would be to get a real, limited priv globus user, but depending on live services during tests is usually a bad idea.)

@sushma-ananda : If you have any interest in this, assign yourself, or if not I can try.

### PR DESCRIPTION
**Pull request by mccalluc**
Chuck is creating a PR for Sushma's work to make it easier to review. Fix #98.

### CODE REVIEW

File: hubmap/frontend/cypress.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "blacklistHosts": ["localhost:8000"],
```
Comment (mccalluc) [inline hubmap/frontend/cypress.json]: This may work, but I think we'd be better off not depending on Django in Cypress. This is subjective, but I think it will be better to use cypress just for testing only the client side code, and all API responses we would get are done with fixtures instead.

File: hubmap/frontend/cypress/integration-examples/examples/cypress_api.spec.js [hunk 2]
```diff
@@ -114,7 +114,7 @@ context('Cypress.config()', () => {
     expect(myConfig).to.have.property('baseUrl', null)
     expect(myConfig).to.have.property('defaultCommandTimeout', 4000)
     expect(myConfig).to.have.property('requestTimeout', 5000)
-    expect(myConfig).to.have.property('responseTimeout', 30000)
+    expect(myConfig).to.have.property('responseTimeout', 60000)
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration-examples/examples/cypress_api.spec.js]: This is just example code... It shouldn't need to change? (When the example code should even be checked in is a valid question... I think it's a helpful resource, but if it's distracting, we can get rid of it.)

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 3]
```diff
@@ -32,4 +32,32 @@ describe('HuBMAP', () => {
     // Footer:
     cy.contains('Supported by the NIH Common Fund');
   });
+
+  describe('The Login Page', function () {
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: I would lean towards just adding the 
```
cy.route(`${api}/auth/login/globus`, 'fixture:login.json');
```
to the list above... but on the other hand, if we want to be sure the code is only accessing the API end-points we mean for it to access, this is better. If your intention is to have a tighter, but more verbose test, then this is good.

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 4]
```diff
@@ -32,4 +32,32 @@ describe('HuBMAP', () => {
...
+      cy.route(`${api}/auth/login/globus`, 'fixture:login.json');
+      cy.setCookie('email', 'test@gmail.com')
+    });
+
+    it('sets auth cookie when logging in via form submission', function () {
+      // click login
+      cy.get('login').click()
+  
+      // we should be redirected to / (home)
+      cy.url().should('include', '/')
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: I think `include` would be a substring anywhere in the URL: Use equality, like https://github.com/hubmapconsortium/hubmap-data-portal/pull/96/files#diff-f65b9ef05f60b40bb52f0b693f2c225fR41

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 5]
```diff
@@ -32,4 +32,32 @@ describe('HuBMAP', () => {
...
+
+      cy.get("loggedin-menu").should('exist')
+  
+      // our auth cookie should be present
+      cy.getCookie('email').should('exist')
+  
+      // UI should reflect this user being logged in
+      cy.get('loggedinemail').should('contain', 'test')
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: I think the test could be more broad, and less fragile: test@gmail.com should just be somewhere on the page, right... we don't really care which element it's in?

File: hubmap/frontend/cypress/integration/live_auth_test.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+//https://docs.cypress.io/guides/references/error-messages.html#Cypress-detected-a-cross-origin-error-happened-on-page-load
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/live_auth_test.js]: What is this link for? What question does it answer? (Clarify the code: Response here not needed.)

File: hubmap/frontend/cypress/integration/stub_globus_auth.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+describe('The Login Page', function () {
+    before(() => {
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/stub_globus_auth.js]: Indentation. (Hopefully, eslint catches this going forward.)

File: hubmap/frontend/cypress/integration/stub_globus_auth.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+      Cypress.Cookies.debug(true)
+      cy.server();
+      const api = 'http://localhost:8000/';
+      cy.route(`${api}/api/?format=json`, 'fixture:base.json');
+      cy.route(`${api}/api/colors/?format=json`, 'fixture:colors.json');
+      cy.route(`${api}/api/genes/?format=json`, 'fixture:genes.json');
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/stub_globus_auth.js]: This is getting copied and pasted a lot: `before` is meant to set up a common starting environment for tests, and we should use it that way.

File: hubmap/frontend/cypress/integration/stub_globus_auth.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      cy.get('#loggedin-menu').should('exist');
+  
+      // auth cookie should be present
+      cy.getCookie('email').should('exist');
+  
+      // UI should reflect this user being logged in
+      cy.get('#loggedinemail').should('contain', 'test@gmail.com').should('exist')
+      cy.get('#loggedin-menu').children().get('#logout-menuitem').should('exist')
+      cy.get('#loggedin-menu').children().get('#logout-menuitem')
+      .children().get('#logout').should('exist')
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/stub_globus_auth.js]: Rather than making all these `exist` tests, do something with the elements.

File: hubmap/frontend/cypress/integration/stub_globus_auth.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        cy.visit('/');
+
+      cy.get('#loggedin-menu').should('exist');
+  
+      // auth cookie should be present
+      cy.getCookie('email').should('exist');
+  
+      // UI should reflect this user being logged in
+      cy.get('#loggedinemail').should('contain', 'test@gmail.com').should('exist')
+      cy.get('#loggedin-menu').children().get('#logout-menuitem').should('exist')
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/stub_globus_auth.js]: I would avoid making the tests too dependent on precise paths... Thinking about writing tests based on what the user sees on the screen instead. Like, there should be a "Logout" on the screen, and clicking should do the right thing.

File: hubmap/frontend/cypress/support/commands.js [hunk 11]
```diff
@@ -23,3 +25,22 @@
 //
 // -- This is will overwrite an existing command --
 // Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
+Cypress.Commands.add('stubLoginUser', () => {
+    const log = Cypress.log({
+        name: 'stubLoginUser',
+    })
+    cy.server()
+    .setCookie('email', 'test@gmail.com')
```
Comment (mccalluc) [inline hubmap/frontend/cypress/support/commands.js]: indentation

File: hubmap/frontend/src/components/home/LogInStatus.js [hunk 12]
```diff
@@ -174,11 +174,11 @@ export default function LoggedInStatus() {
...
                     open={Boolean(anchorEl)}
                     onClose={handleClose}>
-                    <MenuItem  > Globus email: {email}
+                    <MenuItem  id="loggedinemail"> Globus email: {email}
                         <AccountCircleSharp className={classes.rightIcon} />
                     </MenuItem>
-                    <MenuItem onClick={handleCloseLogout} >
-                        <a href={API_URL + 'logout/'} style={{ textDecoration: 'none' }}>
+                    <MenuItem onClick={handleCloseLogout} id="logout-menuitem">
+                        <a id="logout" href={API_URL + 'logout/'} style={{ textDecoration: 'none' }}>
```
Comment (mccalluc) [inline hubmap/frontend/src/components/home/LogInStatus.js]: Let's not add all these IDs: What matters is whether "Logout from Globus" is on the screen: test for that.

File: hubmap/frontend/cypress/support/commands.js [hunk 13]
```diff
@@ -23,3 +25,22 @@
...
+    .route('http://localhost:8000/auth/login/globus/', 'fixture:login')
+    .then(()=>{log.snapshot().end()})
```
Comment (mccalluc) [inline hubmap/frontend/cypress/support/commands.js]: Is this necessary? This is short enough, and only used in one place, so I think it's premature to pull it into a utility like this: It only adds another jump when I'm trying to understand the code.

**Comment by mccalluc** [comment]
(Travis is failing on eslint formatting errors.)

File: hubmap/frontend/cypress/integration-examples/examples/cypress_api.spec.js (hunk 2 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration-examples/examples/cypress_api.spec.js]: Well, when redirecting, ideally it is **not** recommended to test <a> tags/redirects outside of domain in cypress. //https://docs.cypress.io/guides/references/error-messages.html#Cypress-detected-a-cross-origin-error-happened-on-page-load
 However, since you did not establish what you preferred in your original issue, I have implemented both. So using redirects, it takes a little more time, depending on network. so I increased it as per Cypress forums discussion.

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 3 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: So hubmap_spec appears to be written by you, however, please feel free to change this. I have just added this during my tests. We can leave this file out when merging.

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 5 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: As I mentioned in one another comment, hubmap_spec.js is implemented by you, we can leave it with your version.

File: hubmap/frontend/cypress/integration/stub_globus_auth.js (hunk 8 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/stub_globus_auth.js]: yes that is the idea behind this since we have bunch of redirects and stubs even. The reason I spent a whole day on this is to **learn** cypress and work, am aware when before is used, that is the intention behind having before. @mccalluc what do you specifically mean by **copied** how do you know if it is **needed** or **copied** blindly as you are indicating ?

File: hubmap/frontend/cypress/integration/stub_globus_auth.js (hunk 10 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/stub_globus_auth.js]: @mccalluc Well, honestly, this is url logout, so as your issue #98 , we were to test globus auth, we should test complete login and logout. The link exists and is in dropdown menu. Can you please test login/logout manually https://data.dev.hubmapconsortium.org/  and let me know why you are not able to find logout ? Please let me know about this, will be glad to fix any bugs if you don't find logout.

File: hubmap/frontend/src/components/home/LogInStatus.js (hunk 12 above)
Comment (sushma-ananda) [reply hubmap/frontend/src/components/home/LogInStatus.js]: @mccalluc the material-ui menus are hidden, until explicitly opened. So <a> tags are not accessible, as they are hidden. But then we also want to click logout and test it as per your issue #98. So we need ids. We are not adding explicitly, some had ids, some did not. so not sure if you noticed, menus are actually trigged by mousemove and opened and clicked. did you get a chance to run these tests ??

File: hubmap/frontend/cypress/support/commands.js (hunk 13 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/support/commands.js]: @mccalluc This is the recommended stub for url redirects to use this. And please do correct me if I am wrong, I am only a day-old in knowing cypress, we need to add command to add a stub for url redirects outside of domain. cy.route/cy.request do not work for url redirects. Sorry but may be it is not clear what you specifically are looking or meaning, can you please help understand a detail ?

File: hubmap/frontend/cypress.json (hunk 1 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress.json]: Refer: https://github.com/hubmapconsortium/hubmap-data-portal/pull/103#discussion_r315256409

Comment (sushma-ananda) [reply hubmap/frontend/cypress.json]: @mccalluc  However, this is removed. But this is this is one of reasons, I requested we need to document this for testing and requested your suggestion on my reply last week. https://github.com/hubmapconsortium/hubmap-data-portal/issues/98

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 5 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: @mccalluc can you pls take a look at this: https://github.com/hubmapconsortium/hubmap-data-portal/pull/103#discussion_r315265347

File: hubmap/frontend/cypress/integration/stub_globus_auth.js (hunk 8 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/stub_globus_auth.js]: yes that is the idea behind this since we have bunch of redirects and stubs even. The reason I spent a whole day on this is to learn cypress and work, am aware when before is used, that is the intention behind having before. @mccalluc it worries me when you imply *copied everywhere* as you seem to be implying blind copy without knowing what before is ? Are you asking how I came up with before or if I understand concept of before at all ?

File: hubmap/frontend/cypress/integration/stub_globus_auth.js (hunk 9 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/stub_globus_auth.js]: Sorry I did not understand this. do you have specific idea on mind to add additional tests for elements ? I will be glad to do so, as you suggest.

File: hubmap/frontend/cypress/support/commands.js (hunk 13 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/support/commands.js]: If not we will just set cookies in stubs without routes. whichever you suggest.

Comment (sushma-ananda) [reply hubmap/frontend/cypress/support/commands.js]: https://github.com/cypress-io/cypress/issues/2621

File: hubmap/frontend/cypress/integration-examples/examples/cypress_api.spec.js (hunk 2 above)
Comment (mccalluc) [reply hubmap/frontend/cypress/integration-examples/examples/cypress_api.spec.js]: I'm still confused: This example code shouldn't even be running. If this change had any effect, something is wrong.

File: hubmap/frontend/src/components/home/Search.js [hunk 14]
```diff
@@ -158,7 +158,7 @@ class SearchBox extends React.Component {
                                                     100% {fill: ${entry[1]}; opacity: 1;}`);
                                                     tissueElement.style.removeProperty("animation");
                                                     tissueElement.addEventListener("animationend", animationEnd);
-                                                    tissueElement.style.setProperty("animation", tissue + 'tissueAnimation' + " 10s linear");
+                                                    tissueElement.style.setProperty("animation", tissue + "tissueAnimation 10s linear");
```
Comment (mccalluc) [inline hubmap/frontend/src/components/home/Search.js]: These are all good, but it will make things easier if each PR stays on topic:
- In the git history, it will be easier to understand if this comes from a PR just about formatting.
- And in reviewing a PR that's already complicated, it's good to keep it as focused as possible.

Comment (sushma-ananda) [reply hubmap/frontend/src/components/home/Search.js]: Honestly, this errors only showed up in this branch, am sure, there is some branch that would have fixed this eslint errors but is not merged into develop branch. So what might have been fixed in one branch might be different, not sure, I see same errors over and over again. I see travis build fails without this fix every time, as we could expect.

Comment (sushma-ananda) [reply hubmap/frontend/src/components/home/Search.js]: develop branch commit #0857f9c1f987a6e9dd995619d8f46282e664de46 auto-fixes this.

File: hubmap/frontend/src/components/home/LogInStatus.js (hunk 12 above)
Comment (sushma-ananda) [reply hubmap/frontend/src/components/home/LogInStatus.js]: As per our recent discussion: it seems question changes to "do we want to test elements that change/loaded on DOM events/user actions ?"

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 15]
```diff
@@ -1,12 +1,29 @@
 /* eslint no-undef: 0 */
 // TODO: Configure eslint to recognize "cy" as a global.
+// We don't click <a> tags here since hrefs redirect to cross-domain urls, which use
+// stubs (to add/remove cookies). HTTP 304 requested urls are not handled by route/request.
 describe('HuBMAP', () => {
   before(() => {
+    Cypress.Cookies.debug(true);
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: I would prefer that debugging be turned on only as needed: Otherwise there's clutter in the logs and it makes it harder to debug real bugs. (I have the same concern with all the JS code.)

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 16]
```diff
@@ -15,7 +32,6 @@ describe('HuBMAP', () => {
     // Header:
     cy.contains('Browse');
     cy.contains('Help');
-    cy.contains('Login');
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: Replace this with an assertion about what is present instead. (And then click on it to expand the menu, and make assertions on the menu contents.)

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 17]
```diff
@@ -31,5 +47,18 @@ describe('HuBMAP', () => {
...
+    // auth cookie should be present
+    cy.getCookie('email').should('exist');
+
+    // UI should reflect this user being logged in
+    cy.get('#loggedinemail').should('contain', 'test@gmail.com').should('exist');
+    cy.get('#loggedin-menu').children().get('#logout-menuitem').should('exist');
+    cy.get('#loggedin-menu').children().get('#logout-menuitem')
+      .children()
+      .get('#logout')
+      .should('exist');
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: As discussed, we should avoid path-based tests: The are fragile, and harder to fix if they do break.

File: hubmap/frontend/cypress/integration/live_auth_test.js [hunk 18]
```diff
@@ -0,0 +1,63 @@
+// ideally it is not recommended to test tags/redirects outside of domain in cypress. we should not be testing this in cypress,
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/live_auth_test.js]: - As discussed, we do not want to check in code that does not run: This file should be removed.
- Preference for capitalizing sentences.

File: hubmap/frontend/cypress/support/commands.js [hunk 19]
```diff
@@ -1,3 +1,5 @@
+import { cyan } from '@material-ui/core/colors';
```
Comment (mccalluc) [inline hubmap/frontend/cypress/support/commands.js]: I don't think this is needed?

File: hubmap/frontend/src/components/Experiments.js [hunk 20]
```diff
@@ -75,7 +75,6 @@ class Experiments extends React.Component {
                   <StyledTableCell align="right">{experiment.image_count ? experiment._image_count : null}</StyledTableCell>
                 </StyledTableRow>
               ))
-
```
Comment (mccalluc) [inline hubmap/frontend/src/components/Experiments.js]: Unless this was required by linting for some reason, we want to avoid edits to unrelated files during a PR. It makes it a bit harder to understand the history of a project.

File: hubmap/frontend/src/components/home/LogInStatus.js [hunk 21]
```diff
@@ -153,21 +153,25 @@ export default function LoggedInStatus() {
   if (email === '' || email === undefined) {
     return (
       <div className={classes.sectionMenuDesktop}>
-        <a href={`${API_URL}auth/login/globus/`} style={{ textDecoration: 'none' }}>
-
-          <Button color={grey[300]} aria-haspopup="true">
+        <a id="globuslogin" href={`${API_URL}auth/login/globus/`} style={{ textDecoration: 'none' }}>
+          <Button color={grey[300]} aria-haspopup="true" id="button-login">
```
Comment (mccalluc) [inline hubmap/frontend/src/components/home/LogInStatus.js]: - Since we shouldn't depend on IDs like this in cypress tests, please remove.
- In the future, if we do need an element ID, please use dashes: `globus-login` not `globuslogin`.

File: hubmap/frontend/src/components/home/Search.js [hunk 22]
```diff
@@ -131,15 +131,15 @@ class SearchBox extends React.Component {
                     this.searchState = store.getState().searchState;
 
                     // extract what we need for gene/tissue from search results
-                    const results = this.searchState.response !== undefined
+                    let results = this.searchState.response !== undefined
```
Comment (mccalluc) [inline hubmap/frontend/src/components/home/Search.js]: - This seems unrelated to this PR?
- `const` is preferred to `let`.

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 17 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: sorry i am not sure we discussed this. but will see what best i could, given that you are an expert in cypress, i am not. thanks.

File: hubmap/frontend/cypress/integration/live_auth_test.js (hunk 18 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/live_auth_test.js]: ohh yes sure, I can delete this. later whenever you guys want to have prod tests, you can always refer this. 

> * As discussed, we do not want to check in code that does not run: This file should be removed.
> * Preference for capitalizing sentences.

oh yes, sure I will delete this, later when you/anyone decide to have prod tests, you can always refer /this discussion can be helpful.

File: hubmap/frontend/src/components/home/Search.js (hunk 22 above)
Comment (mccalluc) [inline hubmap/frontend/src/components/home/Search.js]: I believe I asked why these are changed from `const` to `let`?
- `const` is preferred: for example: https://medium.com/tanda-developers/why-do-we-prefer-const-d805b9dc03be
- These changes seem unrelated to this PR, in any case?

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 23]
```diff
@@ -1,10 +1,29 @@
...
+  });
+
+  after(() => {
+    cy.server()
+      .clearCookie('email');
+    const api = 'http://localhost:8000/';
+    cy.route(`${api}/api/?format=json`, 'fixture:base.json');
+    cy.route(`${api}/api/colors/?format=json`, 'fixture:colors.json');
+    cy.route(`${api}/api/genes/?format=json`, 'fixture:genes.json');
+    cy.visit('/');
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: Again, this adds an extra unnecessary visit to '/' every test. `after` rarely needed. I think you want to check that "Login" is present when the cookie is cleared? Just make a new `it` for that:
```javascript
it('Displays "Login" when cookie is cleared',  () => {
  cy.server().clearCookie('email');
  cy.visit('/');
  cy.contains('Login');
});
```

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 24]
```diff
@@ -29,6 +47,20 @@ describe('HuBMAP', () => {
...
+
+    cy.get('#loggedin-menu').should('exist');
+
+    // auth cookie should be present
+    cy.getCookie('email').should('exist');
+
+    // UI should reflect this user being logged in
+    cy.get('#button-menu').click().then((el) => {
+      assert.include(el[0].textContent, 'Logged in');
+      cy.get('#loggedin-menu').find('li').as('menuitems').then((menuitem) => {
```
Comment (mccalluc) [inline hubmap/frontend/cypress/integration/hubmap_spec.js]: Again, writing tests that depend on element IDs and DOM structure is deprecated: https://docs.cypress.io/guides/references/best-practices.html#Selecting-Elements ... though I think for us going all the way to `cy-data` is too probably far: Makes the tests less readable, for a small decrease in fragility. Our site is small enough that if the wording did change, that would be worth changing the test, I think.

File: hubmap/frontend/src/components/home/Search.js (hunk 22 above)
Comment (sushma-ananda) [reply hubmap/frontend/src/components/home/Search.js]: I did not make these changes in this. are they coming from some other merge (perhaps from develop) ?

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 23 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: we should check this, whether it is a test we would like to have it in long run or not, is different. if we are testing login/logout, let us do thorough tests.

File: hubmap/frontend/src/components/Experiments.js (hunk 20 above)
Comment (sushma-ananda) [reply hubmap/frontend/src/components/Experiments.js]: this is eslint. I am confident you are not reading my responses for your comments. I know this GUI is messy : when a reviewer starts a review, we receive single email, but when a developer responds to review comment, it goes as single email per comment.

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 23 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: well I have been telling i am 2 days old in cypress. And also I asked some questions on your review comments. You are not addressing my questions, in order to avoid this kind of situation. It is really good, if you can be very **specific** in your review comments, given that I , perhaps *we* have established I am new to cypress ?

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 24 above)
Comment (sushma-ananda) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: well I have been telling i am 2 days old in cypress. And also I asked some questions on your review comments. You are not addressing my questions, in order to avoid this kind of situation. It is really good, if you can be very **specific** in your review comments, given that I , perhaps we, have established I am new to cypress ?

File: hubmap/frontend/src/components/home/Search.js (hunk 22 above)
Comment (mccalluc) [reply hubmap/frontend/src/components/home/Search.js]: Not going into the history too deeply, but it looks like it came from a [merge](https://github.com/hubmapconsortium/hubmap-data-portal/blame/c4b263e0f405c2ca71b6e7c92c33aa8ce4b44a29/hubmap/frontend/src/components/home/Search.js#L134). I'm less interested in the history of this than making sure that every change in this PR is moving us in the right direction.

File: hubmap/frontend/src/components/home/LogInStatus.js (hunk 21 above)
Comment (sushma-ananda) [reply hubmap/frontend/src/components/home/LogInStatus.js]: This comment is referring to specific commit. but most recent commit, does not have this ids.

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 23 above)
Comment (mccalluc) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: If we wanted to test code that set or cleared cookies, perhaps, but the cookie state is controlled by our tests. There is no interesting state that would persist across a new visit. We are just interested in whether the display is appropriate when the cookie is present, and when it is absent, and the later case should be a separate test.

File: hubmap/frontend/cypress/integration/hubmap_spec.js (hunk 16 above)
Comment (mccalluc) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: handled above.

File: hubmap/frontend/cypress/integration/hubmap_spec.js [hunk 25]
```diff
@@ -7,13 +11,40 @@ describe('HuBMAP', () => {
...
+    // auth cookie should be present
+    cy.getCookie('email').should('exist');
+
+    // UI should reflect this user being logged in
+    cy.contains('Logged in').click().then(() => {
+      cy.contains('Globus email').should('exist');
+      cy.contains('Logout from Globus').should('exist');
+    });
+
+    cy.clearCookie('email');
```
[the remark this answers is not in the document]
Comment (mccalluc) [reply hubmap/frontend/cypress/integration/hubmap_spec.js]: Added a test after.

**Comment by mruffalo** [comment]
Thanks for reaching a consensus on this and getting it merged!
